### ISSUE TITLE
Android java GRPC: Error: INTERNAL: Missing HTTP status code 

### ISSUE THREAD

**Issue by mandelmonkey** [grpc/grpc-java#5062]
Please answer these questions before submitting your issue.

### What version of gRPC are you using?
1.8.0

### What did you expect to see?

Correct response,

on android maven build I get the following error on response (note I see the correct response on the logs but the error stops it been returned correctly

```

Error: INTERNAL: Missing HTTP status code
                                                       headers: Metadata(strict-transport-security=max-age=31536000)

//code


  String  macaroon = "my macroon";
  
  ManagedChannel channel = OkHttpChannelBuilder.forAddress("btcpayt3.indiesquare.net", 443).build();
    LightningBlockingStub stub = LightningGrpc
        .newBlockingStub( channel)
        .withCallCredentials(new MacaroonCallCredential(macaroon));
 
    GetInfoResponse response = stub.getInfo(GetInfoRequest.getDefaultInstance());

```

**Comment by carl-mastrangelo** [comment]
It looks like you might be hitting a generic HTTP endpoint rather than a gRPC Server.  We don't use HSTS headers.

**Comment by mandelmonkey** [comment]
@carl-mastrangelo hmm could be the server is using a https proxy between the true grpc server and the client connecting, any setting or flag I can set? I got this working with other grpc platforms such as node or ios

**Comment by mandelmonkey** [comment]
ok it worked now on emulator after changing my grpc version from 1.8.0 to 1.16.1, not sure where I got 1.8.0 from as it doesnt exist right?

**Comment by carl-mastrangelo** [comment]
We include as much of the original response as possible, so you may be able to get more debug info from the Status and trailers.   There is a 1.8.0: https://search.maven.org/artifact/io.grpc/grpc-okhttp/1.8.0/jar

**Comment by mandelmonkey** [comment]
I can confirm the the issue was the version 1.8.0, changing to 1.6.1 fixes it so I will close and post my maven pox.xml code for reference

```
<properties>
    <grpc.version>1.16.1</grpc.version>
</properties> 

<dependencies>
    
   <dependency>
        <groupId>io.grpc</groupId>
        <artifactId>grpc-netty</artifactId>
        <version>${grpc.version}</version>
    </dependency>
 
     <dependency>
    <groupId>io.grpc</groupId>
    <artifactId>grpc-okhttp</artifactId>
    <version>${grpc.version}</version>
</dependency>

  <dependency>
    <groupId>io.grpc</groupId>
    <artifactId>grpc-testing</artifactId>
    <version>${grpc.version}</version>
</dependency>



    <dependency>
        <groupId>io.grpc</groupId>
        <artifactId>grpc-protobuf</artifactId>
        <version>${grpc.version}</version>
    </dependency>
    <dependency>
        <groupId>io.grpc</groupId>
        <artifactId>grpc-stub</artifactId>
        <version>${grpc.version}</version>
    </dependency>
       <dependency>
        <groupId>io.grpc</groupId>
        <artifactId>grpc-stub</artifactId>
        <version>${grpc.version}</version>
    </dependency>
   <dependency>
    <groupId>org.apache.httpcomponents</groupId>
    <artifactId>httpclient</artifactId>
    <version>4.3.4</version>
</dependency>
   
```

**Comment by ericgribkoff** [comment]
You were likely hitting this bug with gRPC OkHttp, fixed in our 1.11.0 release: https://github.com/grpc/grpc-java/issues/4261.
